VideoLocker/src/main/java/org/edx/mobile/view/DiscussionAddPostFragment.java

```diff
@@ -150,6 +153,16 @@ public void onNothingSelected(AdapterView<?> parent) {
 
         addPostButton.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
+                Activity activity = getActivity();
+                if (activity != null) {
+                    InputMethodManager iManager = (InputMethodManager) activity.
+                            getSystemService(Context.INPUT_METHOD_SERVICE);
+                    if (iManager.isAcceptingText()) { // verify if the soft keyboard is open
+                        iManager.hideSoftInputFromWindow(
+                                activity.getCurrentFocus().getWindowToken(), 0);
```

Comment:
Could we move this elsewhere? Something like `SoftKeyboardUtils.hide(Activity)`?

Technically, getCurrentFocus and getSystemService can both return null. And I don't see why you'd need to check if the keyboard is open, there's no harm in trying to "hide" it if it's already open. How about this:
```java
public static void hide(final Activity activity) {
        final InputMethodManager imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
        final View view = activity.getCurrentFocus();
        if (view != null && imm != null) {
            imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
        }
    }
```